main.tf

```diff
@@ -1,6 +1,9 @@
 locals {
-  server_properties = join("\n", [for k, v in var.server_properties : format("%s = %s", k, v)])
-  enable_logs       = var.s3_logs_bucket != "" || var.cloudwatch_logs_group != "" || var.firehose_logs_delivery_stream != "" ? ["true"] : []
+  server_properties            = join("\n", [for k, v in var.server_properties : format("%s = %s", k, v)])
+  enable_logs                  = var.s3_logs_bucket != "" || var.cloudwatch_logs_group != "" || var.firehose_logs_delivery_stream != "" ? ["true"] : []
+  enable_client_authentication = var.enable_client_authentication
+  client_authentication        = local.enable_client_authentication ? 1 : 0
```

Comment:
Sorry but I don't understand this code, why if you have a variable rewrite it as local?

Reply:
@angelabad please see comments below.

`enable_client_authentication` is made available to the operator to toggle the conditional creation of resources by means of true/false in the variable definition.

`client_authentication` is making that conditional available to the dynamic block for the certificate authority ARNs.

https://www.terraform.io/docs/language/values/locals.html
https://www.terraform.io/docs/language/expressions/index.html
https://www.terraform.io/docs/language/expressions/conditionals.html